src/db/read.ts

```diff
@@ -62,6 +66,18 @@ export class Read {
   }
 }
 
+async function* scanIndexMap<R>(
+  idx: IndexRead,
+  dagRead: dag.Read,
+  optsInternal: ScanOptionsInternal,
+  convertEntry: (entry: Entry<ReadonlyJSONValue>) => R,
+  onLimitKey?: (inclusiveLimitKey: string) => void,
+): AsyncIterableIterator<R> {
+  yield* await idx.withMap(dagRead, map =>
```

Comment:
same here... maybe we can return a `Promise<AsyncIterableIterator<R>>` here too. I also think that would help with btree.Write.prototype.scan